Source/Widgets/Viewer/Viewer.js

```diff
@@ -53,13 +54,73 @@ function onTimelineScrubfunction(e) {
   clock.shouldAnimate = false;
 }
 
+function getCesium3DTileFeatureDescription(feature) {
+  var propertyNames = feature.getPropertyNames();
+
+  var html = "";
+  propertyNames.forEach(function (propertyName) {
+    var value = feature.getProperty(propertyName);
+    if (defined(value)) {
+      html += "<tr><th>" + propertyName + "</th><td>" + value + "</td></tr>";
+    }
+  });
+
+  if (html.length > 0) {
+    html =
+      '<table class="cesium-infoBox-defaultTable"><tbody>' +
+      html +
+      "</tbody></table>";
+  }
+
+  return html;
+}
+
+function getCesium3DTileFeatureName(feature) {
+  var i;
+  var possibleNames = [];
+  var propertyNames = feature.getPropertyNames();
+  for (i = 0; i < propertyNames.length; i++) {
+    var propertyName = propertyNames[i];
+    if (/^name$/i.test(propertyName)) {
+      possibleNames[0] = feature.getProperty(propertyName);
+    } else if (/name/i.test(propertyName)) {
+      possibleNames[1] = feature.getProperty(propertyName);
+    } else if (/^title$/i.test(propertyName)) {
+      possibleNames[2] = feature.getProperty(propertyName);
+    } else if (/^(id|identifier)$/i.test(propertyName)) {
+      possibleNames[3] = feature.getProperty(propertyName);
+    } else if (/element/i.test(propertyName)) {
+      possibleNames[4] = feature.getProperty(propertyName);
+    } else if (/(id|identifier)$/i.test(propertyName)) {
+      possibleNames[5] = feature.getProperty(propertyName);
+    }
+  }
+
+  var length = possibleNames.length;
+  for (i = 0; i < length; i++) {
+    var item = possibleNames[i];
+    if (defined(item) && item !== "") {
+      return item;
+    }
+  }
+  return "Unnamed Feature";
+}
+
 function pickEntity(viewer, e) {
   var picked = viewer.scene.pick(e.position);
   if (defined(picked)) {
     var id = defaultValue(picked.id, picked.primitive.id);
     if (id instanceof Entity) {
       return id;
     }
+
+    if (picked instanceof Cesium3DTileFeature) {
+      return new Entity({
+        name: getCesium3DTileFeatureName(picked),
+        description: getCesium3DTileFeatureDescription(picked),
+        feature: picked,
```

Comment:
`feature` doesn't appear to be a documented property of `Entity` ? Should it be?

It is really useful that I can do `viewer.selectedEntity.feature` to get the selected feature so I want to keep that.

Reply:
There's really no good way that I can think of to document it right now.  Entities just support having extra properties added to them, and that's what's happening here.  I'll write up an issue to better document this area of the code (Because we should probably do something similar for ImageryLayer as well, or longer term we should explore `selectedItem` instead of `selectedEntity` and just store the Cesium3DFeature directly on the property instead of "faking' the entity.

Reply:
Actually, I just added an explanation in `selectedEntity` documentation for now, no need to get overly fancy (though we should still improve this later)